src/rust/ide/view/src/searcher/icons.rs

```diff
@@ -0,0 +1,376 @@
+    let head        = Triangle(head_width,head_length).translate_y((-head_length/2.0).px());
+    (line + head).into()
+}
+
+
+
+// =================
+// === Favorites ===
+// =================
+
+/// A five-pointed star.
+mod star {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+            let shape = FiveStar(7.0.px(),0.447);
+            let shape = shape.fill(style.get_color(theme::favorites));
+            shape.shrink(SHRINK_AMOUNT.px()).into()
+        }
+    }
+}
+
+
+
+// ==========
+// === IO ===
+// ==========
+
+/// A rounded rectangle with an arrow pointing in from the left.
+mod data_input {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+
+            // === Rectangle ===
+
+            let rect = Rect((10.0.px(),13.0.px())).corners_radius(1.5.px()).translate_x(1.5.px());
+            // Taking just an outline.
+            let rect = &rect - rect.shrink(1.0.px());
+            // Creating a gap for the arrow to pass through.
+            let gap  = Rect((2.0.px(),4.0.px())).translate_x((-3.0).px());
+            let rect = rect - gap;
+
+
+            // === Arrow ===
+
+            let arrow = arrow(11.0,1.0,4.0,5.0).rotate((PI/2.0).radians()).translate_x(4.0.px());
+
+
+            // === Shape ===
+
+            let shape = rect + arrow;
+            let shape = shape.fill(style.get_color(theme::io));
+            shape.shrink(SHRINK_AMOUNT.px()).into()
+        }
+    }
+}
+
+/// A rounded rectangle with an arrow pointing out to the right.
+mod data_output {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+
+            // === Rectangle ===
+
+            let rect = Rect((9.0.px(),13.0.px())).corners_radius(1.5.px()).translate_x((-2.5).px());
+            // Taking just an outline.
+            let rect = &rect - rect.shrink(1.0.px());
+            // Creating a gap for the arrow to pass through.
+            let gap  = Rect((2.0.px(),4.0.px())).translate_x((1.5).px());
+            let rect = rect - gap;
+
+
+            // === Arrow ===
+
+            let arrow = arrow(11.0,1.0,4.0,5.0).rotate((PI/2.0).radians()).translate_x(8.0.px());
+
+
+            // === Shape ===
+
+            let shape = rect + arrow;
+            let shape = shape.fill(style.get_color(theme::io));
+            shape.shrink(SHRINK_AMOUNT.px()).into()
+        }
+    }
+}
+
+/// A cursor shape, looking roughly like a capital "I".
+fn cursor() -> AnyShape {
+    let middle = Rect((1.0.px(),15.0.px()));
+    let top    = Rect((5.0.px(),1.0.px())).translate_y(7.5.px());
+    let bottom = Rect((5.0.px(),1.0.px())).translate_y((-7.5).px());
+    (middle + top + bottom).into()
+}
+
+/// A rounded rectangle with the letter "A" and a text cursor.
+mod text_input {
+    use super::*;
+
+    ensogl::define_shape_system! {
+        (style:Style) {
+
+            // === Rectangle ===
```

Comment:
This is a border, right? If so, rename it to border pls (the same in other icons)